README.md

```diff
@@ -30,111 +30,259 @@ Recommended Setup
 
 Before you can start, you have to download the browscap.ini file and convert it into a cache. There are two ways.
 
-1. Download the file and convert it in two steps. The downloaded file will be stored in a local file, but there is no check
+a. Download the file and convert it in two steps. The downloaded file will be stored in a local file, but there is no check
    if the remote file has changed. If your cache gets corrupted you only need to rerun the `convert` command.
 
-```php
-vendor/bin/browscap-php browscap:fetch
-vendor/bin/browscap-php browscap:convert
-```
+    ```php
+    vendor/bin/browscap-php browscap:fetch
+    vendor/bin/browscap-php browscap:convert
+    ```
 
-2. Download the file and convert it in one step. The downloaded file will not be stored in a local file, but there is a check
+b. Download the file and convert it in one step. The downloaded file will not be stored in a local file, but there is a check
    if the remote file has changed. If your cache gets corrupted you have clean the cache and restart the process.
 
+    ```php
+    vendor/bin/browscap-php browscap:update
+    ```
+    
+If you want to autoupdate the used cache, we recommend a separate cron job that calls the command listed above.
+
+What's changed in version 3.x
+-----------------------------
+
+##Changes
+
+* the namespace was changed to `BrowscapPHP` 
+* the `Browscap` class was splitted into pieces
+  * for caching the [WurflCache](https://github.com/mimmi20/WurflCache) package is used
+  * for downloading the browscap.ini the [FileLoader](https://github.com/mimmi20/FileLoader) package is used
+
+##removed features
+
+* the autoupdate function was removed
+* all public properties were removed
+
+##new features
+
+* now it is possible to use other caches than the file cache
+* now it is possible to write your own formatter to change the output format 
+* now it is possbile to set a logger
+
+Setup Examples
+--------------
+
+##update your setup to version 3.x
+
+This is the base setup in version 2.x.
 ```php
-vendor/bin/browscap-php browscap:update
+// setup in version 2.x
+$bc = new \phpbrowscap\Browscap($cacheDir);
 ```
 
-If you only want to check if a new version of the browscap.ini is available, you can use the `browscap:check-update` command.
+Change this to the base setup for version 3.x.
+```php
+$bc      = new \BrowscapPHP\Browscap();
+$adapter = new \WurflCache\Adapter\File(array(\WurflCache\Adapter\File::Dir => $cacheDir));
```

Comment:
Let's use short array syntax in examples, especially as min version is 5.6 now anyway